resources/preparing_for_a_security_review.md

```diff
@@ -0,0 +1,10 @@
+- [ ] Enable and address all compiler/linter warnings
```

Comment:
We can move this into development-guidelines, and add a bit of introduction.

Ex: `The following checklist ...`

We can also link to our blogpost for completion